src/main/scala/cromwell/webservice/CromwellApiService.scala

```diff
@@ -1,68 +1,47 @@
 package cromwell.webservice
 
-import akka.actor.{Actor, ActorRef, ActorRefFactory, Props}
+import akka.actor.{Props, Actor, ActorRef}
 import com.typesafe.config.Config
 import cromwell.engine.workflow.{ValidateActor, WorkflowOptions}
 import cromwell.engine.{WorkflowId, WorkflowSourceFiles}
-import lenthall.spray.{SwaggerUiInfo, ConfigSwaggerUiHttpService}
+import lenthall.spray.SwaggerUiResourceHttpService
+import lenthall.spray.WrappedRoute._
+import lenthall.config.ScalaConfig._
 import spray.http.StatusCodes
 import spray.json._
 import spray.routing.Directive.pimpApply
 import spray.routing._
 
-import scala.reflect.runtime.universe._
 import scala.util.{Failure, Success, Try}
 
+trait SwaggerService extends SwaggerUiResourceHttpService {
+  override def swaggerServiceName = "cromwell"
 
-object SwaggerService {
-  /*
-    Because of the implicit arg requirement apply() doesn't work here, so falling back to the less
-    idiomatic (but not unheard of) from().
-   */
-  def from(conf: Config)(implicit actorRefFactory: ActorRefFactory): SwaggerService = {
-    new SwaggerService(conf.getConfig("swagger"))
-  }
-}
-
-class SwaggerService(override val swaggerUiConfig: Config)
-                    (implicit val actorRefFactory: ActorRefFactory)
-  extends ConfigSwaggerUiHttpService {
+  override def swaggerUiVersion = "2.1.1"
 }
 
-
 object CromwellApiServiceActor {
-  def props(workflowManagerActorRef: ActorRef, swaggerService: SwaggerService): Props = {
-    Props(new CromwellApiServiceActor(workflowManagerActorRef, swaggerService))
+  def props(workflowManagerActorRef: ActorRef, config: Config): Props = {
+    Props(classOf[CromwellApiServiceActor], workflowManagerActorRef, config)
   }
 }
 
-class CromwellApiServiceActor(val workflowManager: ActorRef, swaggerService: SwaggerService) extends Actor with CromwellApiService {
+class CromwellApiServiceActor(val workflowManager: ActorRef, config: Config)
+  extends Actor with CromwellApiService with SwaggerService {
   implicit def executionContext = actorRefFactory.dispatcher
   def actorRefFactory = context
 
-  def possibleRoutes = options { complete(StatusCodes.OK) } ~ docsRoute ~ swaggerService.swaggerUiRoutes ~ workflowRoutes
+  def possibleRoutes = workflowRoutes.wrapped("api", config.getBooleanOr("api.routeUnwrapped")) ~ swaggerUiResourceRoute
```

Comment:
Sure, nothing in lenthall is written in stone!

The `get___Or` method names were inspired by the Typesafe Slick team's extension of the Typesafe Config:
https://github.com/slick/slick/blob/3.1.0/slick/src/main/scala/slick/util/GlobalConfig.scala#L50